dts/bindings/led_strip/worldsemi,ws2812-uart.yaml

```diff
@@ -0,0 +1,70 @@
+# Copyright (c) 2024 Croxel, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  Worldsemi WS2812 LED strip, UART binding
+
+  Driver bindings for controlling a WS2812 or compatible LED
+  strip with a Uart.
+
+  The Uart driver should be usable as long as a zephyr Uart API driver
+  is available for your board and has tx-invert property available and
+  enabled.
+  Hardware specific tuning is required using these properties:
+
+    - uart-baudrate
+
+  Data serialisation explanation for WS2812:
+  3 pulses/bits of UART are used to represent one bit of RGB data.
+  During the development of this driver, I was using an ESP32C3 MCU
+  with its UART baud rate set to 2.5Mbits per sec. Hence, 1 bit equals 400ns.
+  For WS2812 timing details, please refer to its datasheet.
+  Link: https://cdn-shop.adafruit.com/datasheets/WS2812.pdf
+  We can represent logic 1 with 0b110 (first two bits high,
+  remaining one bit low),
+  and logic 0 with 0b100 (first bit high, remaining two bits low).
+  We also need to invert the polarity of UART pins because in default,
+  the idle TX line is high, but WS2812 or similar LEDs need the idle
+  line to be low, otherwise, it will misinterpret.
+  When we make the TX line inverted, the start bit will be low to high
+  and the stop bit will be high to low.
+  Hence, the nine bits we are sending in one go will always have 1 at
+  the start and 0 at the end, we just need to calculate middle 7 bits.
+  Also, keep in mind that UART always sends the MSB first.
+  Keeping all this data in mind, I created a lookup table to ease the
+  calculations.
+  This function is just taking three bits at a time from the 24-bit color
+  data, i.e., color[3] array (MSB first),
+  and passing those three bits into the lookup table ws2812_uart_lut[] to
+  retrieve the 7-bit data and fills out
+  the serialized output buffer.
+  This driver can be used with other ic of different timing requirements
+  by simply changing baudrate, bits per rgb bit and lookup table.
+
+
+compatible: "worldsemi,ws2812-uart"
+
+include: [uart-controller.yaml, ws2812.yaml]
+
+properties:
+  bits-per-rgb-bit:
+    type: int
+    required: true
+    description: |
+      Total number of bits representing single
+      RGB data bit.
+
+  lookup-table:
+    type: array
+    required: true
+    description: |
+      Lookup data that will be used to encode
+      RGB data bits into uart streams.
+
+      For example, when setting this driver for
+      WS2812, used 7 uart data bits and 3 bits
+      to represent single RGB data bit, below
+      table was used:
+         lookup-table = <0x5B 0x1B 0x53
+                         0x13 0x5A 0x1A
+                         0x52 0x12>;
```

Comment:
This RGB UART driver is primarily designed for use with small microcontrollers that have limited speed and memory. Therefore, it is optimized to send multiple RGB bits in a single UART transaction. However, as is often the case, optimizing for performance can come at the cost of simplicity.

Reply:
@simonguinot Performance and simplicity topic is debatable, but the deal-breaker issue that @AyushKot96 forgot to mention is the UART speed. That is limited on MCUs as opposed to I2S/SPI.
for e.g. UART speed on MCUs with TX-Invert Property. (Typical values)
ESP32 => 5Mbps, Low end STM32L0 => e.g. L051/053 => 4Mbps.

If we try to send one-frame and or zero-frame, with 1 start + 7 data +1 stop bits then clock speed requirement will ~7.2Mbps for WS2812b (9bit make up 1.25us pulse). That will be an issue for MCU that do not support such high speed.
However with current implementation it is possible to provide a lookup table only having 0 and 1 frame value, with bits_per_rgb bit set to 9.

I have tried driver sample on nucleo_u575 with following overlay file and it work fine for me, @AyushKot96 please add this overlay file in your PR to serve as additional sample
```
#include <zephyr/dt-bindings/gpio/gpio.h>
#include <zephyr/dt-bindings/led/led.h>

#include <zephyr/dt-bindings/led/led.h>

&usart3 {
	status = "okay";
	current-speed = <7200000>;
	pinctrl-0 = < &usart3_tx_pc10 &usart3_rx_pc11 >;
	pinctrl-names = "default";
	data-bits = <7>;
	tx-invert;

	led_strip: ws2812 {
		compatible = "worldsemi,ws2812-uart";
		/* WS2812 */
		chain-length = <30>; /* arbitrary; change at will */
		color-mapping = <LED_COLOR_ID_GREEN
				 LED_COLOR_ID_RED
				 LED_COLOR_ID_BLUE>;
		bits-per-rgb-bit = <9>;		
		lookup-table = <0x7C 0x60>;
	};
};

/ {
	aliases {
		led-strip = &led_strip;
	};
};
```

Let me know if you think otherwise or we can resolve this comment.
CC: @ubieda

Reply:
> @simonguinot Performance and simplicity topic is debatable, but the deal-breaker issue that @AyushKot96 forgot to mention is the UART speed. That is limited on MCUs as opposed to I2S/SPI. for e.g. UART speed on MCUs with TX-Invert Property. (Typical values) ESP32 => 5Mbps, Low end STM32L0 => e.g. L051/053 => 4Mbps.

Hi @cx-anuj-pathak, @AyushKot96 already convinced me with the memory usage optimization.

> 
> If we try to send one-frame and or zero-frame, with 1 start + 7 data +1 stop bits then clock speed requirement will ~7.2Mbps for WS2812b (9bit make up 1.25us pulse). That will be an issue for MCU that do not support such high speed. However with current implementation it is possible to provide a lookup table only having 0 and 1 frame value, with bits_per_rgb bit set to 9.

Well, this was not the idea. The idea was to send a single RGB bit per UART frame. Only 3 or 4 bits (depending on the LED controller model and specification) of the UART frame are used to encode one RGB bit. The remaining bits are set to 0 and ignored by the LED controller. So it doesn't impact the UART speed. This allows to represent zero and one RGB frame in DT as we do for SPI. For users, it is a lot simpler to build than a `lookup-table`. But this also wastes a non negligible amount of memory, which is not great...

> 
> I have tried driver sample on nucleo_u575 with following overlay file and it work fine for me, @AyushKot96 please add this overlay file in your PR to serve as additional sample. (FYI, bytes are sent in LSB first order)

Thanks for testing.